src/stravalib/protocol.py

```diff
@@ -200,7 +209,12 @@ def exchange_code_for_token(
             "expires_at": response["expires_at"],
         }
         self.access_token = response["access_token"]
-        return access_info
+
+        if return_athlete:
+            athlete_info = SummaryAthlete.model_validate(response["athlete"])
```

Comment:
This should happen in the client method

Reply:
ok i'm resolving things that i've addressed!